pandas/core/_numba/executor.py

```diff
@@ -8,15 +8,61 @@
 
 if TYPE_CHECKING:
     from pandas._typing import Scalar
+    from typing import Any
 
 import numpy as np
 
 from pandas.compat._optional import import_optional_dependency
 
 
 @functools.cache
+def make_looper(func, result_dtype, nopython, nogil, parallel):
+    if TYPE_CHECKING:
+        import numba
+    else:
+        numba = import_optional_dependency("numba")
+
+    @numba.jit(nopython=nopython, nogil=nogil, parallel=parallel)
+    def column_looper(
+        values: np.ndarray,
+        start: np.ndarray,
+        end: np.ndarray,
+        min_periods: int,
+        *args,
+    ):
+        result = np.empty((values.shape[0], len(start)), dtype=result_dtype)
+        na_positions = {}
+        for i in numba.prange(values.shape[0]):
+            output, na_pos = func(
+                values[i], result_dtype, start, end, min_periods, *args
+            )
+            result[i] = output
+            if len(na_pos) > 0:
+                na_positions[i] = np.array(na_pos)
+        return result, na_positions
+
+    return column_looper
+
+
+default_dtype_mapping: dict[np.dtype, Any] = {
```

Comment:
Curious, could we not just define [signatures](https://numba.pydata.org/numba-doc/latest/reference/types.html#numba-types) for `numba.jit` to use when running the function?

Reply:
We allocate arrays inside the function and need to pass a dtype there as well. 

Not sure how to access the signature from inside the func.